### DISCUSSION TITLE
PESTPP-MOU with risk? Why is risk maximized?

### DISCUSSION THREAD

**Discussion by RyanConway91** [usgs/pestpp#278] [Q&A]
I am referencing the GMDSI notebook examples for a PESTPP-MOU application I am developing. 

Why is the risk objective maximized? Wouldn't we want to minimize it?

Or does it not matter if its maximized or minimized? Will MOU seek solutions which trade off high and low risk with the other objectives regardless of the optimization direction of risk?

**Comment by jtwhite79** [comment]
You are right that conceptually we want to "minimize risk".  But the way the risk argument is used in pestpp (for legacy reasons....) is such that risk = 0.95 is 95% confidence of satisfying a constraint and risk = 0.05 is 5% confidence of satisfying a constraint.  A better way to think of the risk argument is probably as a reliability index.  So if we maximize the risk argument, we are essentially maximizing reliability...

Typically maximizing risk is in competition with the other objectives, just by the nature of economic and environmental conditions we are seeking, so there will be a pareto front along the risk objective.

**Reply by RyanConway91** [reply]
Thanks, that makes sense. 

If I want decvars (well pumping and injection rates) which satisfied my constraint (no particles can escape the capture system) for **all** the parameters in my risk stack (from IES), would I specify a risk of 1.0? Would I use this as a constraint instead of an objective (I want the particles capture across all my IES ensembles)? 

Another question.... If I start with a big enough initial population size (say like 4* number of decision variables specified in ++mou_population_schedule, then switch back so something more like 2* number of decision variables for subsequent runs) would it be safe to use:
pst.pestpp_options["opt_chance_points"] = "all"
pst.pestpp_options["opt_recalc_chance_every"] = 9999 #reuse initial 

This would give me big population space to map to without needed to rerun the risk stack for every generation?

**Reply by jtwhite79** [reply]
Those are good questions.  If you are using risk as an objective, then you just need make sure 0.001 < risk < 0.999 and risk != 0.5.  MOU will adjust risk itself.  If you arent using risk as an objective, then you have to be careful that your risk value doesnt cause infeasibility.  

"all" chance points is the most rigorous but also the most costly.  And how to use the chance points is very problem specific because it depends on how the chance process interacts with the decvar optimization.  But yeah what you are proposing sounds reasonable, at least as starting point...

**Reply by RyanConway91** [reply]
Thanks! So to clarify, if I want to use a specified risk and not use risk as an objective, I just remove _ risk _ from  mou_objectives? Everything else stays the same (It still needs to be in opt_dec_var_groups?)?

**Reply by jtwhite79** [reply]
I see.  If you arent doing risk as an obj, then mou should ignore that `__risk__` parameter/decvar (or at least not treat it in a special way) so you can mark it as "fixed".  To not do risk as an objective, you need to specify `mou_risk_objective` as false (which is the default value) and then remove `__risk__` from the `mou_objectives` list.

**Reply by RyanConway91** [reply]
Got it. This has been very helpful, thank you
